crates/holochain_integrity_types/src/zome.rs

```diff
@@ -6,13 +6,27 @@
 //! using Rust closures, and is useful for quickly defining zomes on-the-fly
 //! for tests.
 
+use std::borrow::Cow;
+
 use holochain_serialized_bytes::prelude::*;
 
 /// ZomeName as a String.
 #[derive(Clone, Debug, Serialize, Hash, Deserialize, Ord, Eq, PartialEq, PartialOrd)]
-#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 #[repr(transparent)]
-pub struct ZomeName(pub String);
+pub struct ZomeName(pub Cow<'static, str>);
+
+#[cfg(feature = "arbitrary")]
+impl<'a> arbitrary::Arbitrary<'a> for ZomeName {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        Ok(Self(String::arbitrary(u)?.into()))
+    }
+}
+
+/// A trait for converting a value to a [`ZomeName`].
+pub trait ToZomeName {
+    /// Converts the value to a [`ZomeName`].
+    fn zome_name(&self) -> ZomeName;
+}
```

Comment:
Unfortunately no because `Into<ZomeName>` takes `into(self)` which means you can't pass a reference. Come to think of it maybe it could be `AsRef<ZomeName>`

Reply:
Yeh I remember the issue now. What we need is something that takes a `&self` but returns an owned `ZomeName` otherwise you run into lifetime issues. I can't do `T: Into<ZomeName>` because then T must be a reference which usually can't be turned into an `Entry`.

Reply:
Honestly this whole thing is a bit silly because we strictly only ever need a reference to a type that turns into an `Entry` because serialization only needs a reference but I think @thedavidmeister designed the api to be able to take `Foo` or `&Foo` as `T`. I was simply trying to up hold this same constraint. So far I don't know of anyway that a function can take a non-static `T` and then take a reference to that `T`.